mailbox/tika/src/main/java/org/apache/james/mailbox/tika/ContentTypeFilteringTextExtractor.java

```diff
@@ -0,0 +1,53 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.tika;
+
+import java.io.InputStream;
+import java.util.Objects;
+
+import org.apache.james.mailbox.extractor.ParsedContent;
+import org.apache.james.mailbox.extractor.TextExtractor;
+
+public class ContentTypeFilteringTextExtractor implements TextExtractor {
+
+    private final TextExtractor textExtractor;
+    private final TextExtractorConfiguration textExtractorConfiguration;
+
+    public ContentTypeFilteringTextExtractor(TextExtractor textExtractor, TextExtractorConfiguration textExtractorConfiguration) {
+        this.textExtractor = textExtractor;
+        this.textExtractorConfiguration = textExtractorConfiguration;
+    }
+
+    @Override
+    public ParsedContent extractContent(InputStream inputStream, String contentType) throws Exception {
+        if (isBlacklisted(contentType)) {
+            return ParsedContent.empty();
+        }
+        return textExtractor.extractContent(inputStream, contentType);
+    }
+
+    private boolean isBlacklisted(String contentType) {
+        return textExtractorConfiguration
```

Comment:
I would rather create a Set of blacklisted content types because we'll lookup really often, so searching in O(n) looks a bad idea

Reply:
Do you expect that much black listed items?

It is the first time I see you proposing that kind of early ccompute optimization

Reply:
You should read my code more often maybe : we I have a static list/set of things where I constantly do lookup, I usually choose a Set because it's what fits best.
It's not what we could call optimization because the code would be exactly the same but at the List/Set creation point.
Both collections implement `contains` method so `stream()` is overkill in any situation.
BTW a Set is the best representation of what you have, you probably don't want to take care of insertion ordering or duplicate entries, right ?